addons/mail/static/src/core/common/thread_service.js

```diff
@@ -851,45 +852,60 @@ export class ThreadService {
     }
 
     /**
-     * @param {import("models").Persona} persona
-     * @param {import("models").Thread} [thread]
+     * @param {{ persona: import("@mail/core/persona_model").Persona, thread: import("@mail/core/thread_model").Thread }}
      */
-    avatarUrl(persona, thread) {
-        if (!persona) {
-            return DEFAULT_AVATAR;
-        }
-        const urlParams = {};
-        if (persona.write_date) {
-            urlParams.unique = persona.write_date;
-        }
-        if (persona.is_company === undefined && this.store.self?.user?.isInternalUser) {
-            this.personaService.fetchIsCompany(persona);
-        }
-        if (thread?.model === "discuss.channel") {
-            if (persona.type === "partner") {
-                return url(`/discuss/channel/${thread.id}/partner/${persona.id}/avatar_128`, urlParams);
+    avatarUrl({ persona, thread }) {
+        if (thread && !persona) {
+            if (thread.type === "channel" || thread.type === "group") {
+                return url(
+                    `/discuss/channel/${thread.id}/avatar_128`,
+                    assignDefined({}, { unique: thread.avatarCacheKey })
+                );
```

Comment:
This method should be split in two.

It currently returns 2 things that are completely different, the avatar of a thread, and the avatar of a persona inside a channel.

The avatar of the thread can stay named `avatarUrl`.

The avatar of the partner inside a channel should be a separate method.